### PR TITLE
loggerd/encoder: refactor constructors to take EncoderInfo as parameter

### PR DESCRIPTION
**Pull request by deanlee**

### CODE REVIEW

File: system/loggerd/loggerd.h [hunk 1]
```diff
@@ -93,6 +74,7 @@ const EncoderInfo qcam_encoder_info = {
   .frame_height = 330,
   .get_encode_data_func = &cereal::Event::Reader::getQRoadEncodeData,
   .set_encode_idx_func = &cereal::Event::Builder::setQRoadEncodeIdx,
+  .init_encode_data_func = &cereal::Event::Builder::initQRoadEncodeData,
```
Comment (adeebshihadeh) [inline system/loggerd/loggerd.h:77]: might be nice to setup a macro for this so you'd have to specify "QRoadEncod" once

File: system/loggerd/encoderd.cc [hunk 2]
```diff
@@ -54,11 +62,7 @@ void encoder_thread(EncoderdState *s, const LogCameraInfo &cam_info) {
 
       if (buf_info.width > 0 && buf_info.height > 0) {
         for (const auto &encoder_info: cam_info.encoder_infos){
-          encoders.push_back(new Encoder(encoder_info.filename, cam_info.type, buf_info.width, buf_info.height,
-                                        encoder_info.fps, encoder_info.bitrate,
-                                        encoder_info.encode_type,
-                                        encoder_info.frame_width, encoder_info.frame_height,
-                                        encoder_info.publish_name));
+          encoders.push_back(new Encoder(encoder_info, buf_info.width, buf_info.height, encoder_info.encode_type));
```
Comment (adeebshihadeh) [inline system/loggerd/encoderd.cc:65]: remove last arg?

File: system/loggerd/loggerd.h [hunk 3]
```diff
@@ -50,9 +33,11 @@ class EncoderInfo {
   int frame_height = 1208;
   int fps = MAIN_FPS;
   int bitrate = MAIN_BITRATE;
-  cereal::EncodeIndex::Type encode_type = cereal::EncodeIndex::Type::FULL_H_E_V_C;
+  cereal::EncodeIndex::Type encode_type = Hardware::PC() ? cereal::EncodeIndex::Type::BIG_BOX_LOSSLESS
+                                                         : cereal::EncodeIndex::Type::FULL_H_E_V_C;
```
Comment (adeebshihadeh) [inline system/loggerd/loggerd.h:37]: why was this changed?

Comment (adeebshihadeh) [reply system/loggerd/loggerd.h:37]: I see, ffmpeg encoder ignored the input, so it's not changed

Comment (deanlee) [reply system/loggerd/loggerd.h:37]: To remove the last arg codec from `constructor` . `FfmpegEncoder` ignore this arg, and hardcoded it to cereal::EncodeIndex::Type::BIG_BOX_LOSSLESS on pc:


>  FfmpegEncoder(const char* filename, CameraType type, int in_width, int in_height, int fps,
>                 int bitrate, cereal::EncodeIndex::Type **codec**, int out_width, int out_height,
>                 const char* publish_name) :
>                 VideoEncoder(filename, type, in_width, in_height, fps, bitrate, **cereal::EncodeIndex::Type::BIG_BOX_LOSSLESS**, out_width, out_height, publish_name) { encoder_init(); }
>
